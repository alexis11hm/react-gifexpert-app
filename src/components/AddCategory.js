import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {


    const [inputValue, setInputValue] = useState('')

    const handleInputChange = (e) => {
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault()

        if(inputValue.trim().length > 2){
            setCategories(categories => [inputValue,...categories])
            setInputValue('')
        }
    }

    return (
        //si tenemos un elemento que agrupe a otros elementos no es necesario el fragment <></> en este caso, form
        <form onSubmit={handleSubmit}>
                <input 
                    type="text"
                    value={inputValue}
                    onChange={handleInputChange}
                />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
