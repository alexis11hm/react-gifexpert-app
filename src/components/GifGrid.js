import React/*, { useState, useEffect }*/ from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'

import PropTypes from 'prop-types'

export const GifGrid = ({category}) => {

    //data cambian al nombre de images, pero el contenido es lo mismo
    const {data:images, loading} = useFetchGifs(category)

    return (
        <>
            <h3>{category}</h3>
            {loading && <p className="animate__animated animate__flash">Loading</p>}
            <div className="card-grid">
                    {
                        //{..img} es como pasar los atributos de img [id,title, url] y su valor como props, pero de manera corta
                        images.map( (img) => (
                            <GifGridItem 
                                key={ img.id } 
                                { ...img }
                            /> 
                            )
                        )
                    }
            </div>
        </>
    )
}

GifGrid.propTypes ={
    category: PropTypes.string.isRequired
}
