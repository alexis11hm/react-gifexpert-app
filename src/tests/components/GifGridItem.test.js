import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('Pruebas en <GifGridItem />',()=>{

    const title = 'un titulo'
    const url = 'https://localhost/algo.jpg'
    const wrapper = shallow( <GifGridItem title={title} url={url} />)

    test('debe de mostrar el componente correctamente',()=>{
        
        expect(wrapper).toMatchSnapshot()
    })

    test('debe de tener un parrafo con el title',()=>{
        const parrafo = wrapper.find('p')
        expect(parrafo.text().trim()).toBe(title)
    })

    test('debe de tener la imagen igual al url y al de los props',()=>{
        const imagen = wrapper.find('img')
        //prop de la etiqueta img
        expect(imagen.prop('src')).toBe(url)
        expect(imagen.prop('alt')).toBe(title)
    })
    
    test('debe de tener animate_fadeIn en el className',()=>{

        const div = wrapper.find('div')
        const className = div.prop('className')
        expect(className.includes(div.prop('className'))).toBe(true)
    })
})
