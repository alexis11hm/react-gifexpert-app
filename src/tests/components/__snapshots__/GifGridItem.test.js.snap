// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce animate__delay-2s"
>
  <img
    alt="un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    un titulo
  </p>
</div>
`;

exports[`Pruebas en <GifGridItem /> debe de tener un parrafo con el title 1`] = `
<div
  className="card animate__animated animate__bounce animate__delay-2s"
>
  <img
    alt="un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    un titulo
  </p>
</div>
`;
